pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1933,7 +1936,7 @@ public void checkMessageExpiry() {
         int messageTtlInSeconds = topicPolicies.getMessageTTLInSeconds().get();
         if (messageTtlInSeconds != 0) {
             subscriptions.forEach((__, sub) -> {
-                if (!isCompactionSubscription(sub.getName())) {
+                if (systemCursorNames.contains(sub.getName())) {
```

Comment:
I think it would be better to always ignore the compaction subscription. It could be better to change the meaning of the configurable `systemCursorNames` to be additional system cursor names.

Reply:
make sense. Updated.